File: typescript/src/core/modules/pcustomer-management/functions/delete-customer.ts
import { AppError, isStandardError } from "../../../error";
import { PCustomerDAO } from "../data-model/dao";
import { initializeInternalContext } from "../../../context/internal-context";

// Import types
import type { RuntimeContext } from "../../../context/runtime-context";
import type { InternalContext } from "../../../context/internal-context";
import type { TDeletePCustomerParams } from "../data-model/type";

/**
 * Xoá thông tin của một khách hàng ra khỏi danh sách.
 *
 * @param ctx - runtime context.
 *
 * @returns
 */
export async function deleteCustomer(ctx: RuntimeContext) {
  try {
    const params = await ctx.getParams<{ id: string }>();

    const pcustomerDao = new PCustomerDAO();
    const internalCtx =
      initializeInternalContext() as InternalContext<TDeletePCustomerParams>;

    internalCtx.params = {
      query: {
        id: params.id,
      },
    };
    internalCtx.options!.canCatchError = true;

    const result = await pcustomerDao.deletePCustomer(internalCtx);

    return result;
  } catch (error: any) {
    if (isStandardError(error)) return error;

    const err = new AppError("Cannot delete potential customer");
    err.asHTTPError("InternalServerError");
    return err;
  }
}

File: python/src/core/modules/pcustomer_management/functions/delete_customer.py
# Import from core
from core.context.runtime_context import RuntimeContext
from core.context.internal_context.main import initialize_internal_context
from core.error import AppError, is_standard_error
from core.modules.pcustomer_management.data_model.dao import PCustomerDAO


async def delete_customer(ctx: RuntimeContext):
    """
    Xoá thông tin của một khách hàng ra khỏi hệ thống.

    Args:
        ctx (RuntimeContext): Ngữ cảnh runtime chứa thông tin request.

    Returns:
        bool | AppError: Thông tin khách hàng vừa thêm hoặc lỗi chuẩn hóa.
    """
    try:
        params = ctx.get_params()

        pcustomer_dao = PCustomerDAO()
        internal_ctx = initialize_internal_context()

        internal_ctx.params = {"query": {"id": params.get("id")}}
        internal_ctx.options["can_catch_error"] = True

        await pcustomer_dao.delete_pcustomer(internal_ctx)
        return True

    except Exception as error:
        if is_standard_error(error):
            return error

        err = AppError("Cannot delete potential customer")
        err.as_http_error("InternalServerError")
        return err
